src/sfdxProject.ts

```diff
@@ -78,6 +124,42 @@ export class SfdxProjectJson extends ConfigFile<ConfigFile.Options> {
     Object.assign(defaultOptions, options || {});
     return defaultOptions;
   }
+
+  /**
+   * Validates sfdx-project.json against the schema.
+   *
+   * ***See*** [sfdx-project.schema.json] (https://raw.githubusercontent.com/forcedotcom/schemas/master/schemas/sfdx-project.schema.json)
+   */
+  public async schemaValidate(): Promise<void> {
+    if (!this.hasRead) {
+      // read calls back into this method after setting this.hadRead = true
+      await this.read();
+    } else {
+      const validator = new SchemaValidator(this.logger, projectJsonSchema);
+      await validator.load();
+      await validator.validate(this.getContents());
+    }
+  }
+
+  /**
+   * Returns the `packageDirectories` within sfdx-project.json, first reading
+   * and validating the file if necessary.
+   */
+  public async getPackageDirectories(): Promise<PackageDir[]> {
+    // Ensure sfdx-project.json has first been read and validated.
+    if (!this.hasRead) {
+      await this.read();
+    }
```

Comment:
I need to add more validation here.  Since this is a new method I should be able to necessarily validate the file against the schema.